File: thi/example/src/main/java/com/example/model/bus/HomeBus.java
package com.example.model.bus;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "home_bus")
public class HomeBus {
    @Id
    private String id;
    private String name;
    @ManyToOne
    @JoinColumn(name = "type_bus_id", referencedColumnName = "id")
    private TypeBus typeBus;
    @ManyToOne
    @JoinColumn(name = "go_locations", referencedColumnName = "id")
    private Locations goLocations;
    @ManyToOne
    @JoinColumn(name = "back_locations", referencedColumnName = "id")
    private Locations backLocations;
    private String phone;
    private String email;
    private String startTime;
    private String endTime;
}

File: thi/example-2/example/src/app/model/home-bus.ts
import {TypeBus} from "./type-bus";
import {Locations} from "./locations";

export interface HomeBus {
  id: string;
  typeBus: TypeBus;
  name?: string;
  goLocations: Locations;
  backLocations: Locations;
  phone?: number;
  email: string;
  startTime?: string;
  endTime?: string;
}
